Cargo.toml

```diff
@@ -16,6 +16,7 @@ resolver = "2"
 [workspace.dependencies]
 anyhow = "1.0"
 async-trait = "0.1"
+async-std = "1.10"
```

Comment:
I thought we didn't want to introduce an async runtime, which may interfere with the one in uniffi-rs?

Reply:
IINM the stuff in `async-std` should work in a async context because it's just a `Future`, but I might have that wrong?

Reply:
I'm not questioning whether the crate (or the `task::sleep` async function) would work in this library. I remember Oguz mentioned (maybe in one of the PR reviews) that we don't want to introduce another async runtime (like `tokio`) to this library. `async-std` is just another async runtime, like `tokio`. @oguzkocer Please correct me if I am wrong about this.

Reply:
I don't have any experience using `async-std` either, but afaik `async-std` is a little different from `tokio` because it can be run on other executors, including `tokio`. Furthermore, both of them are sort of like ecosystems where some parts of them are related to executors (more so in `tokio`) and some of them are `futures` utilities.

In my opinion, in `wp_api`, we should try to use `futures` as much as possible. I wouldn't be against adding a dependency for `futures` related utilities, but I think it's best to avoid `async-std`, `tokio` etc as much as possible.

Reply:
Does `futures` have a `sleep` function? Its API doc says there is no timer API.

I did a little bit of digging. The `task::sleep` function internally calls [`async_io::Timer::after`](https://docs.rs/async-io/latest/async_io/struct.Timer.html#method.after). Maybe we can use that function directly?

BTW, unlike `futures`, this function uses its own thread instead of the existing async runtime.

> The first time [Async](https://docs.rs/async-io/latest/async_io/struct.Async.html) or [Timer](https://docs.rs/async-io/latest/async_io/struct.Timer.html) is used, a thread named “async-io” will be spawned. - [API doc](https://docs.rs/async-io/latest/async_io/index.html#implementation)

Reply:
My understanding is that `sleep` is a run-time dependent action unless we put the whole thread to sleep, which is not what we want. Considering we are working with different platforms and don't want to rely on anything runtime dependent, I think the easiest and for sure method we can use here is to add a `sleep` function to `RequestExecutor` trait and let each platform provide their own implementation. This should be super easy to implement in all 3 platforms where we have specific runtime executors and it'll keep `wp_api` runtime agnostic.

At some point we might want to extract this new `sleep` function to a new utility async trait where we might add more helpers. However, I don't think there is any need to immediately introduce a new trait.